### PR TITLE
feat(storage): compaction group support table_option of internal_table

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
compaction group support table_option of internal_table

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- TableOption proto 
- register table_option to compaction_group
- interface for table_option read

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
- https://github.com/singularity-data/risingwave/issues/3298

### CODE REVIEW

File: proto/hummock.proto [hunk 1]
```diff
@@ -158,6 +158,12 @@ message CompactionGroup {
   uint64 id = 1;
   repeated bytes member_prefixes = 2;
   CompactionConfig compaction_config = 3;
+
+  message TableOption {
+    uint32 ttl = 1;
+  }
+
+  map<uint32, TableOption> table_options = 4;
```
Comment (skyzh) [inline proto/hummock.proto:166]: The TableOption doesn't look generic enough. In this case, maybe define a single field of `ttl` would be better?

**Review by skyzh — COMMENTED** [review]
Rest LGTM

File: proto/hummock.proto (hunk 1 above)
Comment (skyzh) [inline proto/hummock.proto:166]: may add comments here, table id -> table options mapping

**Review by skyzh — APPROVED** [review]
I think the `ttl` table option is duplicated with the TTL in table catalog. We can refactor later to directly fetch such options from catalog when creating compaction jobs.

File: proto/hummock.proto (hunk 1 above)
Comment (Li0k) [reply proto/hummock.proto:166]: ok, i will use `table_id_to_option` to replace `table_options `

**Review by zwang28 — APPROVED** [review]

**Pull request merged**